### ISSUE TITLE
How to use Streams.stream_to_buffer()

### ISSUE THREAD

**Issue by Quving** [pytube/pytube#1144]
Hello,
I have googled a lot, but I couldn't find any snippet that shows me how to use stream_to_buffer(). I would like to download the audio/video to memory instead of writing it to disk. 

stream_to_buffer() is the way to go, right? 

Thanks for help!

**Comment by johnslavik** [comment]
Yeah. You need to pass there a `BytesIO` object that will be written within each progress. To do this, you need to create the `BytesIO` using the standard `io` library and remember about some charms of using Python streams.

Quick snippet*: 

```python
import io
import pytube

[…]
buf = io.BytesIO()
stream.stream_to_buffer(buf)
buf.seek(0)
data = buf.read()
```

\* Where `stream` is the `pytube.Stream` object you are operating on.
The data, which would be at default written to a file, is now in the `data` label.
Didn’t test, I hope it works.

**Comment by vaccarieli** [comment]
Hi, I am having difficulty with the `pytube.Stream` argument I am passing a the argument `monostate: pytube.monostate.Monostate`, but `stream: Dict[KT, VT]` I am not really sure where do I get that object, for what it seems it is a dictionary, and have tried multiple things from the documentation to get it.

Can u help?

```  File "W:\Synology\ytDownloadPytube.py", line 14, in <module>
    stream = Stream(monostate=Monostate)
TypeError: Stream.__init__() missing 1 required positional argument: 'stream'```
